src/worker/main.ts

```diff
@@ -1,24 +1,65 @@
 import { readExcel } from "./read-wordlist";
-import { SaveFileEventPayload } from "@common/events";
+import {
+  EventPayload,
+  SaveFileRequestPayload,
+  CompileModelRequestPayload,
+  CompileModelResponsePayload,
+  EventType,
+} from "@common/events";
+import {
+  compileModelFromLexicalModelSource,
+  WordListFromArray,
+} from "@predictive-text-studio/lexical-model-compiler";
 import Storage from "./storage";
 
 const storage = new Storage();
 
-// Print the function's name (even though that's self-evident) just to get the
-// linter and Rollup to acknowledge that readExcelSync() is used.
-self.postMessage(`hello, from worker! I can use ${readExcel.name}`);
-
-const handleSaveFileEvent = async (event: MessageEvent) => {
-  const payload = event.data as SaveFileEventPayload;
+const handleEvent = async (event: MessageEvent) => {
+  const payload = event.data as EventPayload;
+  switch (payload.type) {
+    case EventType.SaveFileRequest: {
+      await handleSaveFileEvent(payload as SaveFileRequestPayload);
+    }
+    case EventType.CompileModelRequest: {
+      await handleCompileModel(payload as CompileModelRequestPayload);
+    }
+  }
+};
 
+const handleSaveFileEvent = async (payload: SaveFileRequestPayload) => {
   try {
     await storage.saveFile(payload.name, await payload.file.arrayBuffer());
+    const response: EventPayload = { type: EventType.SaveFileResponse };
+    postMessage(response);
   } catch (e) {
-    postMessage("Save Failed");
-    return;
+    const response: EventPayload = {
+      type: EventType.SaveFileResponse,
+      errorMessage: "Save Failed",
+    };
+    postMessage(response);
   }
+};
 
-  postMessage("Save Success");
+const handleCompileModel = async (payload: CompileModelRequestPayload) => {
+  try {
+    const buffer = await payload.file.arrayBuffer();
+    const wordlist = await readExcel(buffer);
```

Comment:
Not important, but these tasks can be done in parallel by using [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled)

Reply:
Why? readExcel is dependent on the buffer

Reply:
Oh derp, you're right, I didn't notice that >.<